docs/source/deprecation.md

```diff
@@ -1,17 +1,515 @@
-## Deprecation
+# Deprecation
+
+The deprecation process is here to make (your) life of developers easier.
+It's not possible to write perfect solution for everything and as the project
+is evolving, it happens that some functionality needs to be changed, replaced
+or dropped completely. Such situations are inevitable. To reduce negative
+impact on your code, we introduce the deprecation process described below.
+
+## What is covered by deprecation process in leapp?
+
+In short, leapp entities that are supposed to be used by other developers.
+That means e.g.:
+
+1. Models
+1. Shared library classes and functions in leapp repository
+1. Public APIs
+1. Actors providing functionality that could be used by any developer (produce
+   or consume messages)
+
+In other words, private classes, private functions or anything in private
+libraries, may be modified or removed without the deprecation process. As well,
+it's possible we will need to change something (e.g. a behaviour of a function)
+that will not be possible to cover reasonably by the deprecation process (e.g.
+change output of the function...). We'll try our best to prevent it, but it may
+happen. To limit such problems, we recommend people to use APIs as much
```

Comment:
Let's add hyperlink to the APIs...

Reply:
Discussed in person. That's not one specific API, it's generic (one is provided by the framework, one through shared libraries in a repository, another will be provided by workflowAPIs,...)